### PR TITLE
fix edge case for validate json method

### PR DESCRIPTION
**Pull request by ylwu-amzn**
### Description
Fix some edge case can't be handled: `"[]\""`
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by zane-neo — APPROVED** [review]

File: common/src/test/java/org/opensearch/ml/common/utils/StringUtilsTest.java [hunk 1]
```diff
@@ -33,6 +38,10 @@ public void isJson_False() {
         Assert.assertFalse(StringUtils.isJson("{\"key\": \"value}"));
         Assert.assertFalse(StringUtils.isJson("{\"key\": \"value\", \"key\": 123}"));
         Assert.assertFalse(StringUtils.isJson("[1, \"a]"));
+        Assert.assertFalse(StringUtils.isJson("[]\""));
```
Comment (model-collapse) [inline common/src/test/java/org/opensearch/ml/common/utils/StringUtilsTest.java:41]: Need to add cases like "[ ]" and "[, ]".

**Review by model-collapse — DISMISSED** [review]

**Review by model-collapse — APPROVED** [review]

**Review by zane-neo — APPROVED** [review]
